politeiawww/api/cms/v1/v1.go

```diff
@@ -433,3 +446,37 @@ type PaymentInformation struct {
 	AmountReceived  dcrutil.Amount `json:"amountreceived"`
 	Status          PaymentStatusT `json:"status"`
 }
+
+// CMSUser represents a CMS user. It contains the standard politeiawww user
+// fields as well as CMS specific user fields.
+type CMSUser struct {
+	User               www.User        `json:"user"`
+	Domain             DomainTypeT     `json:"domain"` // Contractor domain
+	GitHubName         string          `json:"githubname"`
+	MatrixName         string          `json:"matrixname"`
+	ContractorType     ContractorTypeT `json:"contractortype"`
+	ContractorName     string          `json:"contractorname"`
+	ContractorLocation string          `json:"contractorlocation"`
+	ContractorContact  string          `json:"contractorcontact"`
+	SupervisorUserID   string          `json:"supervisoruserid"`
+}
+
+// UserDetails fetches a cms user's details by their id.
+type UserDetails struct {
+	UserID string `json:"userid"` // User id
+}
+
+// UserDetailsReply returns a cms user's details.
+type UserDetailsReply struct {
+	User *CMSUser `json:"user"`
+}
+
+// EditUser edits a user's preferences.
+type EditUser struct {
+	CMSUser   CMSUser `json:"user"`
+	Signature string  `json:"signature"` // Signature of raw user json
```

Comment:
The user is already authenticated for this route.  What's the purpose of including a signature?

Reply:
Yah I was thinking since the user would be providing a payload of sorts that we should ensure that it's signed.  But if you think that it is unnecessary I can remove.

Reply:
Not necessary.  Signatures are required for data that's being store in politeiad.  User authentication is sufficient for updating user data.

Reply:
Great. Removed.